src/applications/RunMachineShop.java

```diff
@@ -0,0 +1,70 @@
+package applications;
+
+public class RunMachineShop {
+
+    /** process all jobs to completion
+     * @param simulationResults*/
+    static void simulate(SimulationResults simulationResults) {
+        while (JobManager.numJobs > 0) {// at least one job left
+            int nextToFinish = MachineShopSimulator.eList.nextEventMachine();
+            MachineShopSimulator.timeNow = MachineShopSimulator.eList.nextEventTime(nextToFinish);
+            // change job on machine nextToFinish
+            Job theJob = JobManager.changeState(nextToFinish);
+            // move theJob to its next machine
+            // decrement numJobs if theJob has finished
+            if (theJob != null && !MachineShopSimulator.moveToNextMachine(theJob, simulationResults))
+                JobManager.numJobs--;
+        }
+    }
+
+    public static SimulationResults runSimulation(SimulationSpecification specification) {
+        MachineShopSimulator.largeTime = Integer.MAX_VALUE;
+        MachineShopSimulator.timeNow = 0;
+        startShop(specification); // initial machine loading
+        SimulationResults simulationResults = new SimulationResults(JobManager.numJobs);
+        simulate(simulationResults); // run all jobs through shop
+        outputStatistics(simulationResults);
+        return simulationResults;
+    }
+
+    /** output wait times at machines
+     * @param simulationResults*/
+    static void outputStatistics(SimulationResults simulationResults) {
+        simulationResults.setFinishTime(MachineShopSimulator.timeNow);
+        simulationResults.setNumMachines(MachineShopSimulator.numMachines);
+        MachineShopSimulator.setNumTasksPerMachine(simulationResults);
+        MachineShopSimulator.setTotalWaitTimePerMachine(simulationResults);
```

Comment:
Again here, perhaps setNumTasksPerMachine and setTotalWaitTimePerMachine should be pulled into RunMachineShop or MachineShopSimulator should keep outputStatistics.

The second option makes more sense to me as outputting statistics is not strictly part of running the shop